### PR TITLE
Complete TODOs

### PR DESCRIPTION
**Pull request by maniksurtani**
Could @embark or @nerdrew PTAL?

### CODE REVIEW

File: quotaservice-cli/cli.go [hunk 1]
```diff
@@ -39,6 +45,7 @@ var (
 	// update
 	update          = app.Command("update", "Updates namespaces or buckets from a running configuration.")
 	updateGDB       = update.Flag("globaldefault", "Updates the global default bucket.").Short('g').Default("false").Bool()
+	updateFile      = update.Flag("file", "File from which to read configs.").Short('f').String()
```
Comment (embark) [inline quotaservice-cli/cli.go]: This looks like it conflicts with addFile

Comment (maniksurtani) [inline quotaservice-cli/cli.go]: Yeah it's allowed as long as it's on a different instance of command.

**Comment by embark** [comment]
Feature request: show a diff of changes to config, ask for confirmation

**Comment by embark** [comment]
Otherwise LGTM.

File: quotaservice-cli/cli.go [hunk 2]
```diff
@@ -65,38 +72,106 @@ func main() {
...
+		// Global default bucket
+		checkField("name", config.DefaultBucketName, js, func(val string) {
+			kingpin.Fatalf("Global default bucket cannot have name '%v'", val)
+		})
+		checkField("namespace", config.GlobalNamespace, js, func(val string) {
+			kingpin.Fatalf("Global default bucket cannot have namespace '%v'", val)
+		})
+	} else if bucket == "" {
+		// We're just updating a namespace.
+		checkField("name", namespace, js, func(val string) {
```
Comment (embark) [inline quotaservice-cli/cli.go]: are you intentionally checking name against namespace?

File: quotaservice-cli/cli.go [hunk 3]
```diff
@@ -65,38 +72,106 @@ func main() {
...
+			kingpin.Fatalf("Attempting to configure namespace '%v' but config provided is for '%v'", namespace, val)
+		})
+	} else {
+		checkField("name", bucket, js, func(val string) {
```
Comment (embark) [inline quotaservice-cli/cli.go]: are you intentionally checking name against bucket?

**Comment by maniksurtani** [comment]
Added #19

File: quotaservice-cli/cli.go (hunk 2 above)
Comment (maniksurtani) [inline quotaservice-cli/cli.go]: Yes.

File: quotaservice-cli/cli.go (hunk 3 above)
Comment (maniksurtani) [inline quotaservice-cli/cli.go]: Yes.

File: quotaservice-cli/cli.go (hunk 2 above)
Comment (maniksurtani) [inline quotaservice-cli/cli.go]: The JSON config of a namespace contains a `name` element, which refers to the "name of the namespace".

File: quotaservice-cli/cli.go (hunk 3 above)
Comment (maniksurtani) [inline quotaservice-cli/cli.go]: The JSON config of a bucket contains a `name` element which refers to the "name of a bucket".

**Pull request merged**